examples/chatGPT-nft-minter/suave/ChatNFT.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Suave} from "suave-std/suavelib/Suave.sol";
+import {ChatGPT} from "suave-std/protocols/ChatGPT.sol";
+import {Emitter} from "./712Emitter2.sol";
+
+/// @title ChatNFT: ChatGPT-based NFT Creator
+/// @dev ChatNFT is responsible for querying ChatGPT on behalf of the user,
+/// and storing the user's latest query result.
+/// @dev Once the user is satisfied with the query result, they can choose to
+/// create a new NFT with the query result. This can be an image, text;
+/// any bytes-encoded data.
+contract ChatNFT {
+    /// Require function to be called via confidential compute request.
+    modifier confidential() {
+        require(Suave.isConfidential(), "must call confidentially");
+        _;
+    }
+
+    struct MintNFTConfidentialParams {
+        bytes privateKey;
+        address recipient;
+        string[] prompts;
+        string openaiApiKey;
+    }
+
+    event QueryResult(bytes result);
+    event NFTCreated(uint256 tokenId, address recipient, bytes signature);
+
+    function newTokenId(string[] memory prompts) private view returns (uint256) {
+        return uint256(keccak256(abi.encode(prompts, block.timestamp, msg.sender)));
+    }
+
+    /// Logs the query result.
+    function onMintNFT(bytes memory queryResult, uint256 tokenId, address recipient, bytes memory signature) public {
+        emit QueryResult(queryResult);
+        emit NFTCreated(tokenId, recipient, signature);
+    }
+
+    /// Makes a query to ChatGPT with prompts given via confidentialInputs.
+    /// Mints an NFT with the query result.
+    function mintNFT() public confidential returns (bytes memory suaveCalldata) {
```

Comment:
This function need to be called through the confidential request, right?
E.g using the Go SDK, through framework.SendConfidentialRequest()

If that's correct, does this actually work? I'm building something and have similar setup where I call a function using a confidential request, and the function will create a new instance of ChatGPT, the thing is on Rigil testnet it is reverting with `panic: confidential request cannot modify state storage`. Locally though when running suave-geth, it does not revet!